components/brave_wallet/browser/swap_service.cc

```diff
@@ -166,78 +166,88 @@ brave_wallet::mojom::BraveSwapFeeResponsePtr GetBraveFeeInternal(
 }
 
 GURL Append0xSwapParams(const GURL& swap_url,
-                        const brave_wallet::mojom::SwapParams& params) {
+                        const brave_wallet::mojom::SwapQuoteParams& params) {
   GURL url = swap_url;
-  if (!params.taker_address.empty()) {
-    url = net::AppendQueryParameter(url, "takerAddress", params.taker_address);
+  if (!params.from_account_id->address.empty()) {
+    url = net::AppendQueryParameter(url, "takerAddress",
+                                    params.from_account_id->address);
   }
```

Comment:
Let's keep it as is but eventually that should be refactored to SwapService being dependent on KeyringService which gives access to mapping account_id -> address.